src/MongoDB.Driver.Core/Core/Misc/ServerVersion.cs

```diff
@@ -0,0 +1,433 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace MongoDB.Driver.Core.Misc
+{
+    internal class ServerVersion : IEquatable<ServerVersion>, IComparable<ServerVersion>
```

Comment:
Why it's internal? It's until the time when we still have SemanticVersion?

Reply:
As a general rule we don't want to make things public unless we need to.

We expect to make it public in a future release.

Reply:
>We expect to make it public in a future release.

this is what I meant. Thanks